crates/sui-core/src/authority/authority_store.rs

```diff
@@ -1074,115 +825,57 @@ impl AuthorityStore {
         self.objects_lock_table.acquire_read_locks(digests).await
     }
 
-    /// Helper function for updating the objects and locks in the state
-    #[instrument(level = "trace", skip_all)]
-    async fn update_objects_and_locks(
+    /// Updates the state resulting from the execution of a certificate.
+    ///
+    /// Internally it checks that all locks for active inputs are at the correct
+    /// version, and then writes objects, certificates, parents and clean up locks atomically.
+    #[instrument(level = "debug", skip_all)]
+    pub async fn update_state(
```

Comment:
This function also looks like not used? Will this function be the place where checkpoint executor updates the DB?